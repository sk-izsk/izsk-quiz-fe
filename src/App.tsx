import React, { Suspense } from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import 'ui-neumorphism/dist/index.css';
import { AppProvider } from './AppProvider';
import { LoadingScreen, NavBar } from './components';
import { AboutScreen, HomeScreen, LoginScreen, QuizScreen, SignUpScreen } from './lazyComponents';

const App: React.FC = () => {
  return (
    <AppProvider>
      <NavBar />
      <Switch>
        <Suspense fallback={<LoadingScreen />}>
          <Route exact path='/'>
            <Redirect to='/home' />
          </Route>
          <Route path='/home' exact component={HomeScreen} />
          <Route path='/about' exact component={AboutScreen} />
          <Route path='/quiz' exact component={QuizScreen} />
          <Route path='/sign-up' exact component={SignUpScreen} />
          <Route path='/login' exact component={LoginScreen} />
        </Suspense>
      </Switch>
    </AppProvider>
  );
};

export default App;
