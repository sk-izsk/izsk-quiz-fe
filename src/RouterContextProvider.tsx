import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';

export interface RouterContextProviderProps {}

const RouterContextProvider: React.FC<RouterContextProviderProps> = ({ children }) => {
  const history: any = createMemoryHistory();
  return <Router history={history}>{children} </Router>;
};

export { RouterContextProvider };
