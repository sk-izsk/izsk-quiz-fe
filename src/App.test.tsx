import { render } from '@testing-library/react';
import React from 'react';
import App from './App';
import { RouterContextProvider } from './RouterContextProvider';

test('App renders correctly', () => {
  const { asFragment } = render(
    <RouterContextProvider>
      <App />
    </RouterContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
