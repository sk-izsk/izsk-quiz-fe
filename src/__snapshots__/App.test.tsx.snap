// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 384h384v-42.666H64V384zm0-106.666h384v-42.667H64v42.667zM64 128v42.665h384V128H64z"
              />
            </svg>
          </span>
        </button>
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          <a
            class="makeStyles-link-6"
            href="/"
          >
            iZsk Quiz
          </a>
        </h6>
      </div>
    </header>
  </div>
  <div
    class="makeStyles-loaderContainer-13"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="__234s  _2YVD0 _NijRA "
      id="_73635_"
      role="progressbar"
      style="width: 109px; height: 109px;"
    >
      <div
        class="_1QMv-"
        style="width: 100px; height: 100px; transform: translate3d(-50%, -50%, 0) rotate(0deg);"
      >
        <svg
          class="_2vn3-"
          viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            class="_53Md9"
            cx="111.11111111111111"
            cy="111.11111111111111"
            r="50"
            style="stroke-width: 10; stroke-dasharray: 314.159; stroke-dashoffset: 314.1592653589793;"
          />
        </svg>
      </div>
      <div
        class="_3Jx6J"
        style="width: 105px; height: 105px;"
      />
      <div
        class="_1iTMT"
        style="width: 74px; height: 74px;"
      />
      <label
        class="_2hjq3"
        style="font-size: 30px;"
      />
    </div>
  </div>
   
</DocumentFragment>
`;
